claims/verifier.go

```diff
@@ -0,0 +1,83 @@
+package claims
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Verifier bundles a number Assertions
+type Verifier struct {
+	Assertions Assertions
+}
+
+// NewVerifier initializes a new verifier
+func NewVerifier() *Verifier {
+
+	return &Verifier{
+		Assertions: make(map[Claim][]Assertion),
+	}
+
+}
+
+// Add adds an Assertion for a given Claim
+func (v *Verifier) Add(c Claim, a Assertion) *Verifier {
```

Comment:
Why not make the verifier immutable? Just take a list of assertions about claims in `NewVerifier`, that way you can't accidentally have a verifier where you forget to add assertions.